src/Lykke.Service.Subscribers.AzureRepositories/Subscribers/SubscriberRepository.cs

```diff
@@ -0,0 +1,57 @@
+﻿using AzureStorage;
+using Lykke.Service.Subscribers.Core.Repositories;
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+
+namespace Lykke.Service.Subscribers.AzureRepositories.Subscribers
+{
+    public class SubscriberRepository : ISubscriberRepository
+    {
+        private readonly INoSQLTableStorage<Subscriber> _tableStorage;
+
+        public SubscriberRepository(INoSQLTableStorage<Subscriber> tableStorage)
+        {
+            _tableStorage = tableStorage;
+        }
+
+        public async Task<ISubscriber> GetAsync(string email, string source)
+        {
+            var partitionKey = Subscriber.GeneratePartitionKey(source);
+            var rowKey = Subscriber.GenerateRowKey(email);
+
+            return await _tableStorage.GetDataAsync(partitionKey, rowKey);
+        }
+
+        public async Task<IEnumerable<ISubscriber>> GetAllAsync(string source)
+        {
+            var partitionKey = Subscriber.GeneratePartitionKey(source);
+            return await _tableStorage.GetDataAsync(partitionKey);
+        }
+
+        public async Task<ISubscriber> CreateAsync(ISubscriber subscriber)
+        {
+            var partitionKey = Subscriber.GeneratePartitionKey(subscriber.Source);
+            var rowKey = Subscriber.GenerateRowKey(subscriber.Email);
```

Comment:
What if the source is different but the email is already exists?

Reply:
It will creata an account